MapboxMobileEvents/MMEEventsManager.h

```diff
@@ -1,56 +1,151 @@
 #import <Foundation/Foundation.h>
-#import "MMEEvent.h"
-#import "MMETypes.h"
 #import <CoreLocation/CoreLocation.h>
 
-NS_ASSUME_NONNULL_BEGIN
+#import "MMETypes.h"
 
+@class MMEEvent;
 @protocol MMEEventsManagerDelegate;
 
+NS_ASSUME_NONNULL_BEGIN
+
+/*! @brief Mapbox Mobile Events Manager */
 @interface MMEEventsManager : NSObject
 
+/*! @brief events manager delegate */
 @property (nonatomic, weak) id<MMEEventsManagerDelegate> delegate;
+
+/*! @brief YES if metrics collection is enabled */
 @property (nonatomic, getter=isMetricsEnabled) BOOL metricsEnabled;
+
+/*! @brief YES if metrics collection is enabled in the simulator */
 @property (nonatomic, getter=isMetricsEnabledInSimulator) BOOL metricsEnabledInSimulator;
+
+/*! @brief YES if metrics collection is enabled when the app is in use */
 @property (nonatomic, getter=isMetricsEnabledForInUsePermissions) BOOL metricsEnabledForInUsePermissions;
+
+/*! @brief YES if debug logging is enabled */
 @property (nonatomic, getter=isDebugLoggingEnabled) BOOL debugLoggingEnabled;
+
+/*! @brief UserAgent base string, in RFC 2616 format
+    @link https://www.ietf.org/rfc/rfc2616.txt */
 @property (nonatomic, readonly) NSString *userAgentBase;
+
+/*! @brief SDK version, in Semantic Versioning 2.0.0 format
+    @link https://semver.org */
 @property (nonatomic, readonly) NSString *hostSDKVersion;
+
+/*! @brief SKU Identifier */
 @property (nonatomic, copy) NSString *skuId;
+
+/*! @brief Mapbox Access Token
+    @link https://account.mapbox.com */
 @property (nonatomic, copy) NSString *accessToken;
+
+/*! @brief baseURL */
 @property (nonatomic, null_resettable) NSURL *baseURL;
+
+/*! @brief accountType */
 @property (nonatomic) NSInteger accountType;
 
 #pragma mark -
 
+/*! @brief Shared Mabpox Mobile Events Manager */
 + (instancetype)sharedManager;
 
-#pragma mark -
+#pragma mark - Exception Free API
 
+/*!
+ @brief designated initilizer
+ @param accessToken Mapbox Access Token
+ @param userAgentBase UserAgent base string, in RFC 2616 format
+ @param hostSDKVersion SDK version, in Semantic Versioning 2.0.0 format
+ @throws no exceptions
```

Comment:
Instead of document when a method doesn’t throw an exception (i.e., `nothrow` in other languages), can we document when a method does throw and exception and the nature of that exception?